src/instance/minecraft/chat-manager.ts

```diff
@@ -101,8 +103,12 @@ export default class ChatManager extends EventHandler<MinecraftInstance> {
         assert(message) // old packet means message exist
         resultMessage = prismChat.fromNotch(message)
       }
-
-      this.onMessage(resultMessage.toString())
+      this.onMessage(
+        this.proceedFiltering(
+          (data as { senderName?: string }).senderName ?? 'unknown sender',
```

Comment:
I assume you chose "unknown sender" to fill the `author` in the event. The event field is documented to be `undefined` if there is no known author for the event.

Reply:
Nuh uh not documented >:(

Reply:
It was implicit to either use `string` or `undefined`. https://github.com/aidn3/hypixel-guild-discord-bridge/blob/d1c82cef5273d600657a8b50fd301d2ddc4e5bb2/src/common/application-event.ts#L277-L280

I'll clarify it then.

Reply:
Can you check #168 and give me a feedback please?